include/caffe/common_layers.hpp

```diff
@@ -178,6 +178,44 @@ class SplitLayer : public Layer<Dtype> {
   int count_;
 };
 
+/* SliceLayer
+  Takes one blobs and slices it along either num or channel dim,
+  outputting the result into as many top blobs as needed.
+*/
+template <typename Dtype>
+class SliceLayer : public Layer<Dtype> {
+ public:
+  explicit SliceLayer(const LayerParameter& param)
+      : Layer<Dtype>(param) {}
+  virtual void SetUp(const vector<Blob<Dtype>*>& bottom,
+      vector<Blob<Dtype>*>* top);
+
+  virtual inline LayerParameter_LayerType type() const {
+    return LayerParameter_LayerType_SLICE;
+  }
+  virtual inline int ExactNumBottomBlobs() const { return 1; }
+  virtual inline int MinTopBlobs() const { return 2; }
```

Comment:
Why not allow for 1 top blob?  I understand that it doesn't make sense to use a `SliceLayer` for that purpose, but if a user asks for it, the layer could act as a pass-through, doing nothing at all.  This makes the layer more general.